geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java

```diff
@@ -468,7 +474,7 @@ private void processOneConnection(Socket socket, long startTime, DataInputStream
         output.flush();
       }
 
-      handler.endResponse(request, startTime);
+      handler.endResponse(request, startTime2);
```

Comment:
I'm ok with the LongSupplier approach - but I wonder if it would have been better to somehow associate the time supplier with the handler (eg, maybe Handler.startResponse() or something). We're getting the time from one place but then handing it off to another.

Reply:
Maybe that's a good idea for some future work... right now the call to `endResponse` results in operations that could be a bit more clear... e.g.
in `LocatorStats` which is one of the places a call to `handler.endResponse` might end up

`public void endLocatorResponse(long startTime) { `
calls the ubiquitous `getStatTime`
    `long took = DistributionStats.getStatTime() - startTime;`

Reply:
This makes sense @upthewaterspout. Oddly though, in `TcpServer` the start time does not typically coincide with `TcpHandler.processRequest()`. The start time is set earlier. So, as you allude, we'd need to add a method to capture the start time.

If we add only a `startResponse()` method, then that wouldn't make sense for starting a request. If we added two methods: `startRequest()`, `startResponse()` then those impls could capture a start time, but then the possibility of bugs arise: to wit, what happens if a caller starts a request, then ends a response or vice versa.

Reply:
Maybe we could add a generic `startXX()` method. Then the ugly problem is implementing that on all the implementation classes:

![image](https://user-images.githubusercontent.com/4002/67712197-c5224b80-f980-11e9-85e1-b379bdb0e670.png)

Reply:
Looking at the various implementations of e.g. `endRequest()` in that hierarchy, there would be value in eliminating the `startTime` parameter since the vast majority of implementations of that method ignore that parameter! Only `GMSLocatorAdapter()` uses that parameter (which it forwards to stats).